python/dgl/distributed/graph_partition_book.py

```diff
@@ -73,11 +75,27 @@ def get_shared_mem_partition_book(graph_name, graph_part):
     '''
     if not exist_shared_mem_array(_get_ndata_path(graph_name, 'meta')):
         return None
-    is_range_part, part_id, num_parts, node_map, edge_map = _get_shared_mem_metadata(graph_name)
+    is_range_part, part_id, num_parts, node_map_data, edge_map_data = \
+            _get_shared_mem_metadata(graph_name)
     if is_range_part == 1:
-        return RangePartitionBook(part_id, num_parts, node_map, edge_map)
+        # node Id ranges and edge Id ranges are stored in the order of node type Ids
+        # and edge type Ids.
+        node_map = {}
+        ntypes = {}
+        node_map_data = pickle.loads(bytes(F.asnumpy(node_map_data).tolist()))
```

Comment:
Don't understand this line. Why convert it to bytes and load it again?

Reply:
# node_map_data and edge_map_data were serialized with pickle and converted into a list of bytes and then stored in a numpy array before being placed in shared memory. To deserialize, we need to reverse the process.

Reply:
i added comments to explain the process.

Reply:
I'm not sure. the data serialized to the byte array can have a fairly complex data structure.